src/quo/theme.cljs

```diff
@@ -59,3 +59,42 @@
   [component]
   (fn [& args]
     (into [:f> f-with-theme component] args)))
+
+(def previous-screen (atom nil))
+
+(def ^:private change-statusbar-color-delay 150)
+
+(def ^:private dark-screens
```

Comment:
Hi @ibrakhil,
Thank you very much for looking into this issue and finding this cool solution. From the video, your shared solution looks promising.

1. We don't need to keep track of each screen which will require light status bar color, we can just manually call the function when needed.
- For theme, please add `rn/set-status-bar-style theme` to `profile. settings/switch-theme-fx`
- For manual changes we have an event `:change-shell-status-bar-style`, please add the corresponding effect and this `set-staus-bar-style` call their. (feel free to rename the event)
2. why `change-statusbar-color-delay 150`? I think there should not be a delay, but better to discuss with the design team.
3. `set-status-bar-style "light-content" true`. Why true? Please can you try with both `animated?` true and false, and see what's the difference and which one should we keep (discuss with the design team.

Reply:
> We don't need to keep track of each screen

We are keeping track of dark theme screens by adding options in `status-im.navigation.screens` file. (For camera we have in `status-im.navigation.options` file for some reason).

Adding `handle-status-bar-screens` as effects inside screen don't look nice to me. I think it should be part of navigation itself.

Reply:
I guess we probably need to keep track, as `navigtion.screens` options are probably not enough and little scattered.

How about, keep track of dark status bar screens but instead of adding `handle-status-bar-screens` as effect in every screen, we make this call.
1. Inside `profile.settings/switch-theme-fx` for theme changes
2. Inside `:set-view-id-fx`

Something like `(rn/set-status-bar-style (if (get screens/dark-screens view-id) "light-content" "dark-content")`. wdyt?

Btw, did you tried using keywords here for "light-content"?

Reply:
> Inside :set-view-id-fx

In this way, we can also get rid of `change-shell-status-bar-style` event.

Reply:
As navigation-bar and status-bar changes go together, it will be really awesome if we can also consider them while applying these solutions.

Reply:
_"(For camera we have in status-im.navigation.options file for some reason)."_ - @Parveshdhull - we should move this to screens. 
What do you mean scattered btw?

Reply:
I agree with Parvesh though we already have this information and should keep it in one place 👍

Reply:
I mean we do have information stored in screens file, but its not clear. Means there is no clear pattern to differentiate just based on passed options.

Some of dark-screen Ibrahem added here, has these options in screens file
`options/transparent-screen-options`, `options/dark-screen`, `options/camera-screen`.

Maybe we can just add some kind of flag or something as prop to differentiate them, `dark? true`

Reply:
Oh I would have thought they all have a prop like that/ resolve to have a prop like that. Either way let's make that happen we should only be tracking this info once and it seems the best place for it is somewhere in navigation/

Reply:
<img width="699" alt="Screenshot 2024-02-12 at 10 39 06" src="https://github.com/status-im/status-mobile/assets/22799766/17d1001a-d5e6-4944-b1a8-5d1583588e3b">
I thought we had that because this already as this is how we are using the `theme` for the theme provider here 👍

Reply:
Thank you for screenshot, I see now, even though they are using different options they all have `:theme :dark` param (except camera screen). We can also add there and just use data stored in screens file.

Reply:
Yep seems we need to clean up camera screen to use a consistent approach